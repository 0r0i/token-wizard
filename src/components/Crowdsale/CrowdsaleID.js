import React from 'react'
import { ButtonCopyToClipboard } from '../Common/ButtonCopyToClipboard'

export const CrowdsaleID = ({ hash, description, extraClassname = '' }) => {
  return (
    <div className={`cs-CrowdsaleID ${extraClassname}`}>
      <div className="cs-CrowdsaleID_InnerContainer">
        <h3 className="cs-CrowdsaleID_Hash">
          <span className="cs-CrowdsaleID_HashText">{hash}</span> <ButtonCopyToClipboard value={hash} />
        </h3>
        <p className="cs-CrowdsaleID_Description">{description}</p>
      </div>
    </div>
  )
}
