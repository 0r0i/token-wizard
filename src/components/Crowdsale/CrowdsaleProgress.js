import React from 'react'

export const CrowdsaleProgress = ({ totalRaisedFunds, ethGoal, tokensClaimedRatio, extraClassName = '' }) => {
  const renderDividers = () => {
    return Array.apply('', Array(9)).map((noUse, index) => (
      <div key={index} className="cs-CrowdsaleProgress_Division" />
    ))
  }

  return (
    <div className={`cs-CrowdsaleProgress ${extraClassName}`}>
      <div className="cs-CrowdsaleProgress_Funds">
        <div className="cs-CrowdsaleProgress_FundsAmount cs-CrowdsaleProgress_FundsAmount-total-raised">
          <h2 className="cs-CrowdsaleProgress_FundsTitle">
            {totalRaisedFunds} <span className="cs-CrowdsaleProgress_NoBreak">ETH</span>
          </h2>
          <p className="cs-CrowdsaleProgress_FundsDescription">Total Raised Funds</p>
        </div>
        <div className="cs-CrowdsaleProgress_FundsAmount cs-CrowdsaleProgress_FundsAmount-goal">
          <h2 className="cs-CrowdsaleProgress_FundsTitle">
            {ethGoal} <span className="cs-CrowdsaleProgress_NoBreak">ETH</span>
          </h2>
          <p className="cs-CrowdsaleProgress_FundsDescription">Goal</p>
        </div>
      </div>
      <div className="cs-CrowdsaleProgress_Bar">
        <div className="cs-CrowdsaleProgress_BarWidth" style={{ width: tokensClaimedRatio + '%' }} />
        {renderDividers()}
      </div>
    </div>
  )
}
