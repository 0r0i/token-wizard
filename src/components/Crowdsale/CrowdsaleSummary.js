import React from 'react'

export const CrowdsaleSummary = ({
  contributorsCount,
  currentRatePerETH,
  endRatePerETH,
  extraClassName = '',
  isMintedCappedCrowdsale,
  startRatePerETH,
  tokensClaimed,
  totalSupply
}) => {
  const isDutchAuctionCrowdsale = !isMintedCappedCrowdsale

  const renderCrowdsaleSummaryItem = (title, description) => {
    return (
      <div className="cs-CrowdsaleSummary_Item">
        <h3 className="cs-CrowdsaleSummary_ItemTitle">{title}</h3>
        <p className="cs-CrowdsaleSummary_ItemDescription">{description}</p>
      </div>
    )
  }

  return (
    <div className={`cs-CrowdsaleSummary ${extraClassName}`}>
      {renderCrowdsaleSummaryItem(tokensClaimed, 'Tokens Claimed')}
      {renderCrowdsaleSummaryItem(contributorsCount, 'Contributors')}
      {isMintedCappedCrowdsale ? renderCrowdsaleSummaryItem(currentRatePerETH, 'Rate (Tokens/ETH)') : null}
      {renderCrowdsaleSummaryItem(totalSupply, 'Total Supply')}
      {isDutchAuctionCrowdsale ? renderCrowdsaleSummaryItem(startRatePerETH, 'Start Rate (Tokens/ETH)') : null}
      {isDutchAuctionCrowdsale ? renderCrowdsaleSummaryItem(currentRatePerETH, 'Rate (Tokens/ETH)') : null}
      {isDutchAuctionCrowdsale ? renderCrowdsaleSummaryItem(endRatePerETH, 'End Rate (Tokens/ETH)') : null}
    </div>
  )
}
