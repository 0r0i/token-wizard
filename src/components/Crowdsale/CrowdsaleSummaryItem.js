import React from 'react'

export const CrowdsaleSummaryItem = ({ title, description }) => {
  return (
    <div className="cs-CrowdsaleSummaryItem">
      <h3 className="cs-CrowdsaleSummaryItem_Title">{title}</h3>
      <p className="cs-CrowdsaleSummaryItem_Description">{description}</p>
    </div>
  )
}
