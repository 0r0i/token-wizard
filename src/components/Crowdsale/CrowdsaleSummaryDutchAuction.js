import React from 'react'
import { CrowdsaleSummaryItem } from './CrowdsaleSummaryItem'

export const CrowdsaleSummaryDutchAuction = ({
  contributorsCount,
  currentRatePerETH,
  endRatePerETH,
  extraClassName = '',
  startRatePerETH,
  tokensClaimed,
  totalSupply
}) => {
  return (
    <div className={`cs-CrowdsaleSummaryDutchAuction ${extraClassName}`}>
      <CrowdsaleSummaryItem title={tokensClaimed} description={'Tokens Claimed'} />
      <CrowdsaleSummaryItem title={contributorsCount} description={'Contributors'} />
      <CrowdsaleSummaryItem title={totalSupply} description={'Total Supply'} />
      <CrowdsaleSummaryItem title={startRatePerETH} description={'Start Rate (Tokens/ETH)'} />
      <CrowdsaleSummaryItem title={currentRatePerETH} description={'Rate (Tokens/ETH)'} />
      <CrowdsaleSummaryItem title={endRatePerETH} description={'End Rate (Tokens/ETH)'} />
    </div>
  )
}
