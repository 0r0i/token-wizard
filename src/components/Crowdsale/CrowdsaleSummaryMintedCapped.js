import React from 'react'
import { CrowdsaleSummaryItem } from './CrowdsaleSummaryItem'

export const CrowdsaleSummaryMintedCapped = ({
  contributorsCount,
  currentRatePerETH,
  extraClassName = '',
  tokensClaimed,
  totalSupply
}) => {
  return (
    <div className={`cs-CrowdsaleSummaryMintedCapped ${extraClassName}`}>
      <CrowdsaleSummaryItem title={tokensClaimed} description={'Tokens Claimed'} />
      <CrowdsaleSummaryItem title={contributorsCount} description={'Contributors'} />
      <CrowdsaleSummaryItem title={currentRatePerETH} description={'Rate (Tokens/ETH)'} />
      <CrowdsaleSummaryItem title={totalSupply} description={'Total Supply'} />
    </div>
  )
}
