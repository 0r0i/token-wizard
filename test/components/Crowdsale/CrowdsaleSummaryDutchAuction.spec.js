import React from 'react'
import render from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-15'
import { configure, mount } from 'enzyme'
import { CrowdsaleSummaryDutchAuction } from '../../../src/components/Crowdsale/CrowdsaleSummaryDutchAuction'

configure({ adapter: new Adapter() })

describe(`CrowdsaleSummaryDutchAuction`, () => {
  it(`should have some .cs-CrowdsaleSummaryItem children`, () => {
    const crowdsaleSummaryDutchAuctionData = {
      contributorsCount: '5',
      currentRatePerETH: '1000',
      endRatePerETH: '1000',
      startRatePerETH: '50',
      tokensClaimed: '50',
      totalSupply: '100'
    }

    const wrapper = mount(
      <CrowdsaleSummaryDutchAuction
        contributorsCount={crowdsaleSummaryDutchAuctionData.contributorsCount}
        currentRatePerETH={crowdsaleSummaryDutchAuctionData.currentRatePerETH}
        endRatePerETH={crowdsaleSummaryDutchAuctionData.endRatePerETH}
        startRatePerETH={crowdsaleSummaryDutchAuctionData.startRatePerETH}
        tokensClaimed={crowdsaleSummaryDutchAuctionData.tokensClaimed}
        totalSupply={crowdsaleSummaryDutchAuctionData.totalSupply}
      />
    )
    const children = wrapper.find('.cs-CrowdsaleSummaryItem')

    expect(Object.keys(crowdsaleSummaryDutchAuctionData).length === children.length).toBeTruthy()
  })

  it(`should render CrowdsaleSummaryDutchAuction component`, () => {
    // Given
    const component = render.create(<CrowdsaleSummaryDutchAuction />)

    // When
    const tree = component.toJSON()

    // Then
    expect(tree).toMatchSnapshot()
  })
})
