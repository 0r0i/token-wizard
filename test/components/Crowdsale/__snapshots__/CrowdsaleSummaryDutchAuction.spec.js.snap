// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CrowdsaleSummaryDutchAuction should render CrowdsaleSummaryDutchAuction component 1`] = `
<div
  className="cs-CrowdsaleSummaryDutchAuction "
>
  <div
    className="cs-CrowdsaleSummaryItem"
  >
    <h3
      className="cs-CrowdsaleSummaryItem_Title"
    />
    <p
      className="cs-CrowdsaleSummaryItem_Description"
    >
      Tokens Claimed
    </p>
  </div>
  <div
    className="cs-CrowdsaleSummaryItem"
  >
    <h3
      className="cs-CrowdsaleSummaryItem_Title"
    />
    <p
      className="cs-CrowdsaleSummaryItem_Description"
    >
      Contributors
    </p>
  </div>
  <div
    className="cs-CrowdsaleSummaryItem"
  >
    <h3
      className="cs-CrowdsaleSummaryItem_Title"
    />
    <p
      className="cs-CrowdsaleSummaryItem_Description"
    >
      Total Supply
    </p>
  </div>
  <div
    className="cs-CrowdsaleSummaryItem"
  >
    <h3
      className="cs-CrowdsaleSummaryItem_Title"
    />
    <p
      className="cs-CrowdsaleSummaryItem_Description"
    >
      Start Rate (Tokens/ETH)
    </p>
  </div>
  <div
    className="cs-CrowdsaleSummaryItem"
  >
    <h3
      className="cs-CrowdsaleSummaryItem_Title"
    />
    <p
      className="cs-CrowdsaleSummaryItem_Description"
    >
      Rate (Tokens/ETH)
    </p>
  </div>
  <div
    className="cs-CrowdsaleSummaryItem"
  >
    <h3
      className="cs-CrowdsaleSummaryItem_Title"
    />
    <p
      className="cs-CrowdsaleSummaryItem_Description"
    >
      End Rate (Tokens/ETH)
    </p>
  </div>
</div>
`;
